Sources/OAuth2Client/NXOAuth2Connection.m

```diff
@@ -418,7 +418,7 @@ - (void)connection:(NSURLConnection *)connection didReceiveResponse:(NSURLRespon
     if (self.statusCode == 401
         && client.accessToken.refreshToken != nil
         && authenticateHeader
-        && [authenticateHeader rangeOfString:@"expired_token"].location != NSNotFound) {
+        && ([authenticateHeader rangeOfString:@"invalid_token"].location != NSNotFound || [authenticateHeader rangeOfString:@"invalid_request"].location != NSNotFound)) {
```

Comment:
`invalid_token` is not listed in the [error codes of the RFC6749](http://tools.ietf.org/html/rfc6749#section-5.2) Curiously `expired_token` also seems wrong (it was correct for [Draft 10](http://tools.ietf.org/html/draft-ietf-oauth-v2-10), when this was built). `invalid_client` seems to be the correct error response together with a 401 (see RFC linked above).